### PR TITLE
ENH: rework how device templates are found

### PR DESCRIPTION
**Pull request by klauer**
Working toward closing #248 

- [x] Search paths for either .ui or .py files, with .py taking precedence
- [x] Test with simple `FakeEpicsMotor`
- [ ] `ui_dir` is set to `typhos/ui`: extend this to multiple directories as in `PYDM_DISPLAYS_PATH` (or perhaps just repurposing it?)

cc @ZLLentz

### CODE REVIEW

File: typhos/utils.py [hunk 1]
```diff
@@ -420,6 +420,15 @@ def no_device_lazy_load():
         Device.lazy_wait_for_connection = old_val
 
 
+def pyqt_class_from_enum(enum):
```
Comment (hhslepicka) [inline typhos/utils.py]: Wouldn't it be best to move this into `pcdsutils` at the `qt` module?

File: typhos/display.py [hunk 2]
```diff
@@ -63,18 +56,23 @@ class TyphosDeviceDisplay(TyphosBase, TyphosDesignerMixin, DisplayTypes):
...
-        self.templates = dict((_typ.name, os.path.join(ui_dir,
-                                                       _typ.name + '.ui'))
-                              for _typ in self.TemplateEnum)
+
+        # Without a device set
+        self.templates = {
+            templ.name: os.path.join(ui_dir, templ.name + '.ui')
+            for templ in self.TemplateEnum
+        }
+        print(self.templates)
```
Comment (hhslepicka) [inline typhos/display.py]: Debug print here

File: typhos/utils.py (hunk 1 above)
Comment (klauer) [reply typhos/utils.py]: sure - once the PR is finished we can move it

Comment (hhslepicka) [reply typhos/utils.py]: Nein... move it now! :neckbeard:

Comment (klauer) [reply typhos/utils.py]: Move pydm to mily

Comment (hhslepicka) [reply typhos/utils.py]: Move your 🍑 to ...

File: typhos/utils.py [hunk 3]
```diff
@@ -418,3 +419,81 @@ def no_device_lazy_load():
...
+        yield f'{name}'
+
+        if not include_mro:
+            break
+
+
+def remove_duplicate_items(list_):
+    'Return a de-duplicated list/tuple of items in `list_`, retaining order'
+    cls = type(list_)
+    return cls(sorted(set(list_), key=list_.index))
```
Comment (hhslepicka) [inline typhos/utils.py]: 🅰️ ➕

**Comment by hhslepicka** [comment]
I like the idea of a search path for Typhos templates following the `PYDM_DISPLAYS_PATH`. This helps us a lot for relocatable displays as well as for `development` and `production` areas.

**Comment by klauer** [comment]
Considering this only a part of the template piece. Merging for now.

**Pull request merged**